### ISSUE TITLE
Custom blueprint action

### ISSUE THREAD

**Issue by behzadeshan** [balderdashy/sails#3138]
I want to handle soft delete/undelete actions in all models. I have overridden destroy action in api/blueprints/destroy.js, hopefully soft delete action is handled now. But I want to handle undelete as well. I have added undelete.js to the same folder as destroy.js, and also I have written a custom route for it in config/routes.js, but in the clinet side I am getting this error "404 (Not Found)", I tried these custom routes but none of them did work:

  'GET /undelete/:id': {
    blueprint: 'undelete'
  }

  'GET /:modelIdentity/undelete/:id': {
    blueprint: 'undelete'
  }

  'GET /:model/undelete/:id': {
    blueprint: 'undelete'
  }

While, when I use the exact name of model in route it works. 
Is it possible to write such a route or I have to write a hook?

**Comment by crh3675** [comment]
Just write the code home-slice:

    'GET /undelete/:id': {
        controller :'someController',
        action  : 'someAction;   
    }

**Comment by behzadeshan** [comment]
@crh3675, your mentioned route is not a blueprint route.

**Comment by sailsbot** [comment]
Thanks for posting, @behzadeshan.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
